lib/docker/api.go

```diff
@@ -51,6 +52,10 @@ type ImageService interface {
 	// Returns the list of images synced
 	Sync(ctx context.Context, dir string, progress utils.Printer) ([]TagSpec, error)
 
+	// SyncForScanning syncs all images in a local directory, with a remote registry that is capable of scanning and
+	// reporting vulnerabilities in those images
+	SyncForScanning(ctx context.Context, conf ScanningConfig, dir string, progress utils.Printer) (err error)
```

Comment:
This interface extension also wouldn't be needed if we just provide an alternative implementation of ImageService.